speechbrain/utils/checkpoints.py

```diff
@@ -75,6 +77,41 @@
 CKPT_PREFIX = "CKPT"
 METAFNAME = f"{CKPT_PREFIX}.yaml"  # Important that this is not .ckpt
 PARAMFILE_EXT = ".ckpt"  # ...because these files will be
+# some keys have been renamed in the new version of the code
+KEYS_MAPPING: Dict[str, str] = {
+    "mutihead_attn": "multihead_attn",  # see PR #2489
+}
+
+
+def map_old_state_dict_weights(
+    state_dict: Dict[str, torch.Tensor], mapping: Dict[str, str]
+) -> Dict[str, torch.Tensor]:
+    """
+    Maps the keys in the old state dictionary according to the provided mapping,
+    under the given prefix.
+
+    Parameters
+    ----------
+    state_dict : dict
+        The old state dictionary to be mapped.
+    mapping : dict
+        A dictionary specifying the mapping between old and new keys.
```

Comment:
It is unclear the extent of what this is, in particular, to my understanding:

- This can affect any part of the key, which could look like `somemodule.somesubmodule.somefield` rather than just `somefield`
- This will do partial matches on keys (e.g. `"a": "b"` will turn `_a` into `_b`)

The behavior should be documented to avoid accidentally including renames with too many side effects.